src/main/scala/tilelink/Buffer.scala

```diff
@@ -82,3 +82,67 @@ object TLBuffer
       .getOrElse(TLNameNode("no_buffer"))
   }
 }
+
+class TLBufferNodeAndNotCancel (
+  a: BufferParams,
+  b: BufferParams,
+  c: BufferParams,
+  d: BufferParams,
+  e: BufferParams)(implicit valName: ValName) extends TLAdapterNodeAndNotCancel(
+    clientFn  = { p => p.v1copy(minLatency = p.minLatency + b.latency + c.latency) },
+    managerFn = { p => p.v1copy(minLatency = p.minLatency + a.latency + d.latency) }
+) {
+  override lazy val nodedebugstring = s"a:${a.toString}, b:${b.toString}, c:${c.toString}, d:${d.toString}, e:${e.toString}"
+
+}
+
+class TLBufferAndNotCancel(
+  a: BufferParams,
+  b: BufferParams,
+  c: BufferParams,
+  d: BufferParams,
+  e: BufferParams)(implicit p: Parameters) extends LazyModule
+{
+  def this(ace: BufferParams, bd: BufferParams)(implicit p: Parameters) = this(ace, bd, ace, bd, ace)
+  def this(abcde: BufferParams)(implicit p: Parameters) = this(abcde, abcde)
+  def this()(implicit p: Parameters) = this(BufferParams.default)
+
+  val node = new TLBufferNodeAndNotCancel(a, b, c, d, e)
+
+  lazy val module = new LazyModuleImp(this) {
+    (node.in zip node.out) foreach { case ((in, edgeIn), (out, edgeOut)) =>
+      out.a <> a(in.a.asDecoupled)
```

Comment:
So a TLBufferAndNotCancel is the adapter that converts us back to standard Decoupled? That makes sense.
Where is the adapter from TLNode to TL_ACancel?

Reply:
I hadn't needed an adapter from TLNode to TL_ACancel yet.  I can add one, but it would be un-tested.